### PR TITLE
backup: write backup status as http response

### PR DESCRIPTION
**Pull request by xiang90**
fix https://github.com/coreos/etcd-operator/issues/1018

### CODE REVIEW

**Comment by hongchaodeng** [comment]
Also update https://github.com/coreos/etcd-operator/blob/master/doc/user/backup_service.md ?

File: pkg/backup/backup.go [hunk 1]
```diff
@@ -154,7 +154,11 @@ func (b *Backup) Run() {
 		lastSnapRev = rev
 
 		if ackchan != nil {
-			ackchan <- err
+			ack := backupNowAck{err: err}
+			if err != nil {
+				ack.status = b.recentBackupsStatus[len(b.recentBackupsStatus)-1]
```
Comment (hongchaodeng) [inline pkg/backup/backup.go]: Maybe add a method `getMostRecentStatus()`?

**Comment by xiang90** [comment]
@hongchaodeng can you give this a manual test? our backup pkg is not testable right now. but i do not want to block @yifan-gu for that.

at the mean time, i will address your comments.

**Comment by hongchaodeng** [comment]
Sure.
Let's do a manual test before we merge it.
We can also think about refactoring the backup struct to make it testable.

**Comment by xiang90** [comment]
@hongchaodeng can you give this a test so that we can merge it in?

**Comment by hongchaodeng** [comment]
Not working:
```
$ curl http://10.96.9.159:19999/v1/backupnow
{"creationTime":"","size":0,"version":"","timeTookInSecond":0}
```

**Comment by xiang90** [comment]
@hongchaodeng try again?

**Comment by hongchaodeng** [comment]
Works now:
```
{"creationTime":"2017-05-08T23:39:33Z","size":0.023,"version":"3.1.4","timeTookInSecond":1}
```

**Comment by hongchaodeng** [comment]
LGTM

**Pull request merged**